exporter/azuremonitorexporter/factory.go

```diff
@@ -52,87 +69,84 @@ func (f *factory) createTracesExporter(
 	set exporter.Settings,
 	cfg component.Config,
 ) (exporter.Traces, error) {
-	exporterConfig, ok := cfg.(*Config)
-
+	f.initLogger(set.Logger)
+	config, ok := cfg.(*Config)
 	if !ok {
 		return nil, errUnexpectedConfigurationType
 	}
-
-	tc, errInstrumentationKeyOrConnectionString := f.getTransportChannel(exporterConfig, set.Logger)
-	if errInstrumentationKeyOrConnectionString != nil {
-		return nil, errInstrumentationKeyOrConnectionString
-	}
-
-	return newTracesExporter(exporterConfig, tc, set)
+	ame := getOrCreateAzureMonitorExporter(cfg, set)
+
+	return exporterhelper.NewTraces(
+		context.TODO(),
```

Comment:
you should use the context passed in line 68

Reply:
resolved